src/sentry/search/events/filter.py

```diff
@@ -1113,8 +1228,7 @@ def resolve_params(self) -> List[WhereType]:
         conditions.append(Condition(self.column("timestamp"), Op.GTE, start))
         conditions.append(Condition(self.column("timestamp"), Op.LT, end))
 
-        # If we already have projects_to_filter, there's no need to add an additional project filter
-        if "project_id" in self.params and len(self.projects_to_filter) == 0:
```

Comment:
This was causing a validation problem with `OR` searches on projects. Snuba is expecting a top-level condition on project as a requirement, but when the project conditions are `OR`ed together, it fails this validation. Removing this to be in sync with the existing JSON syntax.